test/object-store/sync/app.cpp

```diff
@@ -4172,6 +4172,18 @@ TEST_CASE("app: jwt login and metadata tests", "[sync][app][user][metadata][func
 
     SECTION("jwt happy path") {
         bool processed = false;
+        bool logged_in_once = false;
+
+        auto token = app->subscribe([&logged_in_once, &app](auto&) {
+            if (!logged_in_once) {
```

Comment:
Would it make sense to assert here instead?

```suggestion
            REQUIRE(!logged_in_once);
```

Reply:
good point, I didn't have the unsubscribe originally so was expecting it to fire twice with the logout